scripts/rest-api-cognito/account-B-api-gateway.yaml

```diff
@@ -0,0 +1,163 @@
+    Type: Number
+    Default: '29000'
+  
+Resources:
+  ###################
+  ### API Gateway ###
+  ###################
+
+  MyAPIGatewayAuthorizer:
+    Type: AWS::ApiGateway::Authorizer
+    Properties:
+      AuthorizerResultTtlInSeconds: !Ref AuthorizerResultTtlInSeconds
+      IdentitySource: method.request.header.Authorization
+      Name: CognitoAuthorizer
+      ProviderARNs:
+        - !Ref CognitoUserPoolArn
+      RestApiId: !Ref MyAPIGateway
+      Type: COGNITO_USER_POOLS
+
+  MyAPIGateway:
+    Type: AWS::ApiGateway::RestApi
+    Properties:
+      Description: Example API Gateway
+      EndpointConfiguration:
+        Types:  
+          - REGIONAL
+      Name: !Ref ApiGatewayName
+
+  WhoAmIResource:
+    Type: AWS::ApiGateway::Resource
+    Properties:
+      ParentId: !GetAtt MyAPIGateway.RootResourceId
+      PathPart: whoami
+      RestApiId: !Ref MyAPIGateway
+
+  ECSResource:
+    Type: AWS::ApiGateway::Resource
+    Properties:
+      ParentId: !GetAtt MyAPIGateway.RootResourceId
+      PathPart: ecs
+      RestApiId: !Ref MyAPIGateway
+
+  ECSMethod:
+    Type: AWS::ApiGateway::Method
+    Properties:
+      AuthorizationType: COGNITO_USER_POOLS
+      AuthorizerId: !Ref MyAPIGatewayAuthorizer
+      HttpMethod: ANY
+      Integration:
+        ConnectionId: !Ref ApiPrivateLink
+        ConnectionType: VPC_LINK
+        IntegrationHttpMethod: ANY
+        IntegrationResponses:
+          - StatusCode: 200
+            ResponseTemplates:
+              application/json: "{}"
+        PassthroughBehavior: WHEN_NO_TEMPLATES
+        RequestParameters: 
+          integration.request.header.x-pagopa-pn-cx-role: context.authorizer.claims.custom:Role
+        TimeoutInMillis: !Ref TimeoutInMillis
+        Type: HTTP_PROXY
+        Uri: !Ref APIURI
+      MethodResponses: #Sending back headers to client for verification of role headers
+        - StatusCode: 200
+      ResourceId: !Ref ECSResource
+      RestApiId: !Ref MyAPIGateway
+
+  ApiPrivateLink:
+    Type: AWS::ApiGateway::VpcLink
+    Properties: 
+      Description: "API Gateway VPC link for a REST API"
+      Name: !Sub ${ApiGatewayName}-VPC-link
+      TargetArns: 
+        - !Ref NLBArn
+
+  WhoAmIMethod:
+    Type: AWS::ApiGateway::Method
+    Properties:
+      AuthorizationType: COGNITO_USER_POOLS
+      AuthorizerId: !Ref MyAPIGatewayAuthorizer
+      HttpMethod: POST
+      Integration:
+        ConnectionType: INTERNET
+        IntegrationHttpMethod: POST
+        IntegrationResponses:
+          - StatusCode: 200
+            ResponseParameters:
+              method.response.header.x-pagopa-pn-cx-role: context.authorizer.claims.custom:Role
+            ResponseTemplates:
+              application/json: "{ 
+                \"User-Data\" : { 
+                    \"Roles\" : [
+                        \"#foreach($i in $context.authorizer.claims['custom:Role'].split(','))\" {
+                            \"User-Role\": $i
```

Comment:
@KamoKG please update the template, this one doesn't render a correct JSON: https://pagopa.atlassian.net/browse/PN-2593?focusedCommentId=47368